### PR TITLE
ARACHNE-1389

### PR DESCRIPTION
**Pull request by MikMironov**
allow to upload images of diff ext as avatar
still saving with .jpg ext for compatibility with old images

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java [hunk 1]
```diff
@@ -612,8 +608,9 @@ public U removePublicationFromUser(Long userId, Long publicationId) throws NotEx
     public void saveAvatar(U user, MultipartFile file)
             throws IOException, WrongFileFormatException, ImageProcessingException, MetadataException {
 
-        if (!Files.getFileExtension(file.getOriginalFilename()).equalsIgnoreCase(AVATAR_EXT)) {
-            throw new WrongFileFormatException("file", "only JPG");
+        String file_ext = file.getContentType().split("/")[1];
+        if (!file.getContentType().split("/")[0].equals(AVATAR_CONTENT_TYPE)) {
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java]: Use CommonFileUtils

File: src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java [hunk 2]
```diff
@@ -612,8 +608,9 @@ public U removePublicationFromUser(Long userId, Long publicationId) throws NotEx
     public void saveAvatar(U user, MultipartFile file)
             throws IOException, WrongFileFormatException, ImageProcessingException, MetadataException {
 
-        if (!Files.getFileExtension(file.getOriginalFilename()).equalsIgnoreCase(AVATAR_EXT)) {
-            throw new WrongFileFormatException("file", "only JPG");
+        String file_ext = file.getContentType().split("/")[1];
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java]: Why underscore in variable naming?

File: src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java [hunk 3]
```diff
@@ -624,12 +621,14 @@ public void saveAvatar(U user, MultipartFile file)
         if (!userFilesDir.exists()) {
             userFilesDir.mkdirs();
         }
-        final File avatar = Paths.get(userFilesDir.getPath(), "avatar.jpg").toFile();
+        final File avatar = Paths.get(userFilesDir.getPath(), AVATAR_FILE_NAME).toFile();
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java]: How is it going to work with all stuff below? Detection of orientation, image preparation, etc. E.g. there is JpegDirectory class later

File: src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java [hunk 4]
```diff
@@ -137,7 +133,7 @@
 public abstract class BaseUserServiceImpl<U extends User, S extends Skill, SF extends SolrField> implements BaseUserService<U, S> {
     private static final Logger LOGGER = LoggerFactory.getLogger(BaseUserServiceImpl.class);
     private static final String USERS_DIR = "users";
-    private static final String AVATAR_EXT = "jpg";
+    private static final String AVATAR_CONTENT_TYPE = "image";
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java]: this constant is already defined - com.odysseusinc.arachne.commons.utils.CommonFileUtils#TYPE_IMAGE just make it public

**Pull request merged**